### DISCUSSION TITLE
Alpine `x-on:mouseenter` event not working after DOM element deleted using Livewire.

### DISCUSSION THREAD

**Discussion by johncarter-** [livewire/livewire#1377] [4. General]
Firstly, I'm not sure if this is a Livewire or [Alpine ](https://github.com/alpinejs/alpine) issue, so sorry if I need to clear off.

 - I have some CSS grid items that have a `x-on:mouseenter` event attached to them.
 - The grid items have a button that fires a delete event via a Livewire (`wire:click`).
 - When the deleted grid item is removed from the DOM and the next grid item fills its place the `x-on:mouseenter` fails to work on the subsequent grid elements.

Explainer of issue: https://youtu.be/e6hPF5PFMC8

Code:

``` 
<div x-data="{hovered: null}">
  <div x-on:mouseenter="hovered = {{ $task->id }};" x-on:mouseleave="hovered = null" :class="{'border-blue-600 bg-gray-50' : hovered == {{ $task->id }} }">
    <div wire:click="markDone({{ $task->id }})">&times;</div>
  </div>
  <!-- more divs -->
</div>
```

**Comment by tanthammar** [comment]
Hi @johncarter- , I've picked up this topic from the discord channel. (I'm called `@stockholm` there)
Also read the Alpine thread.

Did you add wire:key to your loop?
```blade
<div  x-data="{hovered: null}">
@foreach($tasks as $task)
  <div wire:key="{{ $task->id }}" x-on:mouseenter="hovered = {{ $task->id }};" x-on:mouseleave="hovered = null" :class="{'border-blue-600 bg-gray-50' : hovered == {{ $task->id }} }">
    <div wire:click="markDone({{ $task->id }})">&times;</div>
  </div>
  <!-- more divs -->
@endforeach
</div>
```

Alternatively you can try to add `wire:ignore` to each item where you set `x-data`. The wire:click task-id will still (have not tested :) be set on the first render of the component.
```blade
<div x-data="{hovered: null}">
@foreach($tasks as $task)
  <div  wire:ignore x-on:mouseenter="hovered = {{ $task->id }};" x-on:mouseleave="hovered = null" :class="{'border-blue-600 bg-gray-50' : hovered == {{ $task->id }} }">
    <div wire:click="markDone({{ $task->id }})">&times;</div>
  </div>
@endforeach
  <!-- more divs -->
</div>
```

Or try to add both wire:key and wire:ignore ...

**Comment by tanthammar** [comment]
If none of the above works, and instead of the solution you picked up here #1284, 

you can reinit only this component by adding `x-ref`.

It is better than searching through the entire page.

```blade
<div x-data="{hovered: null}" x-ref="taskList">
  <!-- the loop -->
</div>
```

```js
window.livewire.hook('afterDomUpdate',  () => {
        let el = document.querySelector('[x-ref="taskList"]');
        Alpine.initializeComponent(el);
    });
```

**Comment by johncarter-** [comment]
Thanks for coming back to me @tanthammar really helpful of you.

Yes I did try the `wire:key` as Ryan suggested but it didn't get me anywhere.

I will have to undo the hack that I have at the moment to see if your suggestions help. I am really grateful those and will report back when I get a chance to work on this again ( end of next week hopefully ).
